core/trino-main/src/main/java/io/trino/server/remotetask/DynamicFiltersFetcher.java

```diff
@@ -107,6 +107,12 @@ public synchronized void start()
         fetchDynamicFiltersIfNecessary();
     }
 
+    public synchronized void stop()
+    {
+        // stop fetching dynamic filters while allowing any outstanding request to complete
+        running = false;
```

Comment:
Can stop be invoked before start?  or can the start() be invoked after stop() (for some reason `start()` is indempotent)
then the stop signal would be lost

i think it cannot, but we could make DFF immune to such reordering,
eg change `running` into `AtomicReference<Sate>` where State has 3 values INITIAL, RUNNING, STOPPED